File: 1 semestr/lab6/lab6_3/lab6_3.cpp
#include <iostream>
#include <cmath>
#include <ctime>
#include <cstdlib>
using namespace std;

int** zapolnenie(int count_row, int count_columns) {

	srand(time(NULL));

	int** matrix = new int* [count_row];
	for (int i = 0; i < count_columns; i++) {
		matrix[i] = new int[count_columns];
	}

	for (int row = 0; row < count_row; row++) {
		for (int columns = 0; columns < count_columns; columns++) {
			matrix[row][columns] = rand() % 101 - 50;
			cout << matrix[row][columns] << '\t';
		}
		cout << endl;
	}
	return matrix;
}

int max_stroka(int** matrix) {
	int max_v_str = 0, index_max_str;
	for (int row = 0; row < sizeof(matrix); row++) {
		int proizvedenie = 1;
		for (int columns = 0; columns < sizeof(matrix[0]); columns++) {
			if (matrix[row][columns] < 0) {
				proizvedenie = proizvedenie * matrix[row][columns];
			}
		}
		if (abs(proizvedenie) > max_v_str) {
			max_v_str = abs(proizvedenie);
			index_max_str = row;
		}
	}
	return index_max_str;
}

int main() {
	
	int count_row, count_columns, index;
	int** matrix;

	cout << "Number of lines: ";
	cin >> count_row;
	cout << "Number of columns: ";
	cin >> count_columns;
	cout << endl;

	matrix = zapolnenie(count_row, count_columns);
	index = max_stroka(matrix);

	cout << "Index: " << index;

}

File: 1 semestr/lab6/lab6_3.py
from random import randint
def zapolnenie(strings_array,columns_array):
    matrix = [0] * strings_array
    for i in range(strings_array):
        matrix[i] = [0] * columns_array
        for j in range(columns_array):
            matrix[i][j] = randint(-50,50)
            print (matrix[i][j], end = '\t')
        print ('\n', end = '')
    return matrix

def max_stroka(array):
    max_v_str = 0
    for i in range(len(array)):
        proizvedenie = 1
        for j in range(len(array[0])):
            if array[i][j] < 0:
                proizvedenie *= array[i][j]
        if abs(proizvedenie) > max_v_str:
            max_v_str = abs(proizvedenie)
            index_max_string = i
    return index_max_string

strings_array = int(input("Введите кол-во строк матрицы: "))
columns_array = int(input("Введите кол-во столбцов матрицы: "))

array = zapolnenie(strings_array,columns_array)
index = max_stroka(array)

print ("Индекс строки с маскимальным по модулю произведением элементов: ", index)